app/src/features/Joins/FilterJoinList.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback, useEffect } from "react";
+
+import { Button, Grid, makeStyles, Typography } from "@material-ui/core";
+import AddIcon from "@material-ui/icons/AddCircleOutline";
+import { useTranslation } from "react-i18next";
+import { v4 as uuid } from "uuid";
+
+import { useAppDispatch, useAppSelector } from "app/store";
+import {
+  columnAdded,
+  columnSelectors,
+  columnRemoved,
+  columnUpdated,
+} from "features/Columns/columnSlice";
+import { resourceSelectors } from "features/Mappings/resourceSlice";
+import { Column, Filter } from "services/api/generated/api.generated";
+
+import JoinSelect from "./JoinSelect";
+import { joinAdded, joinRemoved, joinSelectors } from "./joinSlice";
+
+const useStyles = makeStyles(() => ({
+  button: {
+    textTransform: "none",
+  },
+}));
+
+type JoinProps = {
+  filter: Partial<Filter>;
+};
+
+const FilterJoinList = ({ filter }: JoinProps): JSX.Element | null => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const dispatch = useAppDispatch();
+
+  const filterColumn = useAppSelector((state) =>
+    columnSelectors.selectById(state, filter.sql_column ?? "")
+  );
+  const mapping = useAppSelector((state) =>
+    resourceSelectors.selectById(state, filter.resource ?? "")
+  );
+  const columns = useAppSelector((state) => columnSelectors.selectAll(state));
+  const joins = useAppSelector((state) => joinSelectors.selectAll(state));
+  const filterJoins = joins.filter((join) => join.column === filterColumn?.id);
+  const columnsByJoin = (joinId?: string) =>
+    columns.filter((column) => column.join === joinId);
+
+  const handleJoinAdd = useCallback(() => {
+    const joinId = uuid();
+    dispatch(
+      columnAdded({
+        id: uuid(),
+        join: joinId,
+        owner: mapping?.primary_key_owner,
+        table: mapping?.primary_key_table,
+      })
+    );
+    dispatch(
+      columnAdded({
+        id: uuid(),
+        join: joinId,
+        owner: filterColumn?.owner,
+        table: filterColumn?.table,
+      })
+    );
+    dispatch(
+      joinAdded({
+        id: joinId,
+        column: filterColumn?.id,
+      })
+    );
```

Comment:
nice! the only problem will be when we add multiple joins, the pre-fill won't be right but I think it's fine